c10/core/Scalar.h

```diff
@@ -88,6 +88,18 @@ class C10_API Scalar {
 
   Scalar operator-() const;
   Scalar conj() const;
+  Scalar log() const;
+
+  template<typename T>
+  bool equal(T num) const {
```

Comment:
When comparing a Scalar against a C++ number should people always use equal?

Reply:
yeah

Reply:
I am not very familiar with c++ semantic, why is it different from `operator==`?

Reply:
To compare a scalar to a number, we first have to obtain the stored value in the scalar which usually is done by `.toDouble()` but with the possibility that the scalar can also be complex, we can't always do that anymore. With the new `equal` method, that's automatically taken care of

Reply:
Offline conversation: add a special case for bool

Reply:
Ho you cannot template the `operator==` the same way as you do with equal?

Given that using `.equal()` and not `==` is a (small) constraint I'm just trying to understand why we actually need to do it. But this is not blocking for this PR!

Reply:
`operator==` can be overloaded to compare two objects of the same type (in this case, scalars), but not objects of different types (a scalar and a number). If we do add this overload, we'd first have to convert the number to a scalar and then compare it with the scalar object in question. But that, to me, seems like an overkill